Kernel/VM/PhysicalRegion.cpp

```diff
@@ -23,10 +23,11 @@ PhysicalAddress PhysicalRegion::take_free_page()
     if (m_used == m_pages)
         return {};
 
-    for (unsigned page = 0; page < m_pages; page++) {
+    for (unsigned page = m_last; page < m_pages; page++) {
```

Comment:
If we do get to the end of `m_pages`, we might still have some available pages before `m_last`, no?
Shouldn't we wrap around at the end and continue until `page` reaches `m_last - 1`?

Reply:
You're right, it is possible. I'd say _most_ of the time `m_last` will be pointing in the right place, but with some fragmentation and some out of order deallocations, it could be pointing too far ahead.